cglib/src/main/java/net/sf/cglib/core/EmitUtils.java

```diff
@@ -425,7 +425,7 @@ private static void hash_object(CodeEmitter e, Type type, Customizer customizer)
         Label end = e.make_label();
         e.dup();
         e.ifnull(skip);
-        if (customizer != null) {
+        if (customizer != null && !(customizer instanceof Customizer2)) {
```

Comment:
By making this method a no-op in Customizer2, this new checks would not be necessary.